server/router/api.go

```diff
@@ -24,168 +24,172 @@ import (
-	}
+		queue := apiBase.Group("/queue")
+		{
+			queue.Use(session.MustAdmin())
+			queue.GET("/info", api.GetQueueInfo)
+			queue.GET("/pause", api.PauseQueue)
+			queue.GET("/resume", api.ResumeQueue)
+			queue.GET("/norunningpipelines", api.BlockTilQueueHasRunningItem)
+		}
 
-	secrets := e.Group("/api/secrets")
-	{
-		secrets.Use(session.MustAdmin())
-		secrets.GET("", api.GetGlobalSecretList)
-		secrets.POST("", api.PostGlobalSecret)
-		secrets.GET("/:secret", api.GetGlobalSecret)
-		secrets.PATCH("/:secret", api.PatchGlobalSecret)
-		secrets.DELETE("/:secret", api.DeleteGlobalSecret)
-	}
+		secrets := apiBase.Group("/secrets")
+		{
+			secrets.Use(session.MustAdmin())
+			secrets.GET("", api.GetGlobalSecretList)
+			secrets.POST("", api.PostGlobalSecret)
+			secrets.GET("/:secret", api.GetGlobalSecret)
+			secrets.PATCH("/:secret", api.PatchGlobalSecret)
+			secrets.DELETE("/:secret", api.DeleteGlobalSecret)
+		}
 
-	if zerolog.GlobalLevel() <= zerolog.DebugLevel {
-		debugger := e.Group("/api/debug")
+		logLevel := apiBase.Group("/log-level")
 		{
-			debugger.Use(session.MustAdmin())
-			debugger.GET("/pprof/", debug.IndexHandler())
-			debugger.GET("/pprof/heap", debug.HeapHandler())
-			debugger.GET("/pprof/goroutine", debug.GoroutineHandler())
-			debugger.GET("/pprof/block", debug.BlockHandler())
-			debugger.GET("/pprof/threadcreate", debug.ThreadCreateHandler())
-			debugger.GET("/pprof/cmdline", debug.CmdlineHandler())
-			debugger.GET("/pprof/profile", debug.ProfileHandler())
-			debugger.GET("/pprof/symbol", debug.SymbolHandler())
-			debugger.POST("/pprof/symbol", debug.SymbolHandler())
-			debugger.GET("/pprof/trace", debug.TraceHandler())
+			logLevel.Use(session.MustAdmin())
+			logLevel.GET("", api.LogLevel)
+			logLevel.POST("", api.SetLogLevel)
 		}
-	}
 
-	logLevel := e.Group("/api/log-level")
-	{
-		logLevel.Use(session.MustAdmin())
-		logLevel.GET("", api.LogLevel)
-		logLevel.POST("", api.SetLogLevel)
+		apiBase.GET("/signature/public-key", session.MustUser(), api.GetSignaturePublicKey)
+
+		apiBase.POST("/hook", api.PostHook)
+
+		if zerolog.GlobalLevel() <= zerolog.DebugLevel {
+			debugger := apiBase.Group("/debug")
+			{
+				debugger.Use(session.MustAdmin())
+				debugger.GET("/pprof/", debug.IndexHandler())
+				debugger.GET("/pprof/heap", debug.HeapHandler())
+				debugger.GET("/pprof/goroutine", debug.GoroutineHandler())
+				debugger.GET("/pprof/block", debug.BlockHandler())
+				debugger.GET("/pprof/threadcreate", debug.ThreadCreateHandler())
+				debugger.GET("/pprof/cmdline", debug.CmdlineHandler())
+				debugger.GET("/pprof/profile", debug.ProfileHandler())
+				debugger.GET("/pprof/symbol", debug.SymbolHandler())
+				debugger.POST("/pprof/symbol", debug.SymbolHandler())
+				debugger.GET("/pprof/trace", debug.TraceHandler())
+			}
+		}
 	}
 
-	e.GET("/api/signature/public-key", session.MustUser(), api.GetSignaturePublicKey)
-
 	// TODO: remove /hook in favor of /api/hook
 	e.POST("/hook", api.PostHook)
```

Comment:
Do we already use `/api/hook` when adding a project and when repairing?

Reply:
no 😆 

And I personally would not describe this as an "api" function as it's not restfull

I would rm /api/hooks instead

same goes with streams, they are not restufll so we should not prefix them with api

Reply:
Api has nothing to do with REST in my understanding. All of those routes are our interface to get data from and into the backend. Putting those urls under a common prefix makes proxying them much easier as well.

Reply:
what about
/api/* for all backend stuff
and 
/api/v1/* for all restfull stuff?

-> it's about make it easy to generate an openswagger docu and generate clients

else we can just ask what the generall consensus is in mainteiner channel?

Reply:
> -> it's about make it easy to generate an openswagger docu and generate clients

Swagger just ignores not specified / disabled endpoints so that should not be a problem.

Reply:
well I'll merge this refactor ... and we can move on i an dedicated pull for that ...